### DISCUSSION TITLE
how to get api information with sdk js

### DISCUSSION THREAD

**Discussion by luis9430** [directus/directus#2765] [Archive]
Hello, I am new to this and I am creating a project with nuxt and directus, I am trying to connect to the sdk js, I have already made the connection but I cannot get the data to be printed on the front
my code is this

```js
import DirectusSDK from "@ directus / sdk-js";

const client = new DirectusSDK();

client.login ({
   url: "https://localhost/directus/public",
   project: "****",
   email: "*******",
   password: "********",
   persist: false
});

async function fetchAllItems () {
   const data = await client.getItems ("example");
   return data;
}

console.log(fetchAllItems());
```

when I check my console I get this message: "Unauthorized request"

I know it is my mistake, but I am new and I am a bit out of place, I hope you can help me, thanks in advance

**Comment by rijkvanzanten** [comment]
@luis9430 `login`, just like your own function `fetchAllItems()` is async, so you'll have to wait for it to be done before you fire the fetchAllItems function. 

Likewise, you have to wait for the `fetchAllItems` function to be done before you can console log the result:

```js
import DirectusSDK from "@directus/sdk-js";

const client = new DirectusSDK();

async function dataExample() {
  await client.login({ ... });

  const data = await client.getItems('example');

  console.log(data);
}
```

**Reply by luis9430** [reply]
thank you so much, it worked!
